aten/src/ATen/native/mps/operations/BitwiseOps.mm

```diff
@@ -16,103 +16,106 @@
 }}
 
 kernel void bitwise_lshift_tensor(device {0}  *out [[buffer(0)]],
-                         constant {1}  *a [[buffer(1)]],
-                         constant {2}  *b [[buffer(2)]],
+                         constant {0}  *a [[buffer(1)]],
+                         constant {0}  *b [[buffer(2)]],
                          uint offset [[thread_position_in_grid]]) {{
   out[offset] = a[offset] << b [offset];
 }}
 
 kernel void bitwise_lshift_scalar(device {0}  *out [[buffer(0)]],
-                         constant {1}  *a [[buffer(1)]],
-                         constant {2}  &b [[buffer(2)]],
+                         constant {0}  *a [[buffer(1)]],
+                         constant {0}  &b [[buffer(2)]],
                          uint offset [[thread_position_in_grid]]) {{
   out[offset] = a[offset] << b;
 }}
 
 kernel void bitwise_rshift_tensor(device {0}  *out [[buffer(0)]],
-                         constant {1}  *a [[buffer(1)]],
-                         constant {2}  *b [[buffer(2)]],
+                         constant {0}  *a [[buffer(1)]],
+                         constant {0}  *b [[buffer(2)]],
                          uint offset [[thread_position_in_grid]]) {{
   out[offset] = a[offset] >> b [offset];
 }}
 
 kernel void bitwise_rshift_scalar(device {0}  *out [[buffer(0)]],
-                         constant {1}  *a [[buffer(1)]],
-                         constant {2}  &b [[buffer(2)]],
+                         constant {0}  *a [[buffer(1)]],
+                         constant {0}  &b [[buffer(2)]],
                          uint offset [[thread_position_in_grid]]) {{
   out[offset] = a[offset] >> b;
 }}
 
 kernel void bitwise_not(device {0}  *out [[buffer(0)]],
-                         constant {1}  *a [[buffer(1)]],
+                         constant {0}  *a [[buffer(1)]],
                          uint offset [[thread_position_in_grid]]) {{
   out[offset] = ~a[offset];
 }}
 )METAL",
-                              3);
-
-static inline std::string getMetalType(const c10::ScalarType scalar_type) {
-  TORCH_CHECK(c10::isIntegralType(scalar_type, /*includesBool=*/true), "Unsupported type");
-  return scalarToMetalTypeString(scalar_type);
-}
-
-static inline std::string getMetalType(const Tensor& t) {
-  return getMetalType(t.scalar_type());
-}
-
-static inline std::string getMetalType(const c10::Scalar& s) {
-  return getMetalType(s.type());
-}
+                              1);
 
 template <typename ScalarOrTensor>
 static id<MTLComputePipelineState> getCPLState(const Tensor& t1,
                                                const Tensor& t2,
                                                const ScalarOrTensor& t3,
                                                const std::string& fname) {
-  return lib.getPipelineStateForFunc(fname, {getMetalType(t1), getMetalType(t2), getMetalType(t3)});
+  TORCH_CHECK(c10::isIntegralType(t1.scalar_type(), /*includesBool=*/true), "Unsupported type ", t1.scalar_type());
```

Comment:
If these aren't tested extensively yet, many of these could be `TORCH_CHECK_VALUE` instead